src/web/mjs/connectors/To-Corona-Ex.mjs

```diff
@@ -51,30 +51,89 @@ export default class ToCoronaEx extends Connector {
 
     async _getChapters(manga) {
         let chapterList = [];
+        const getToken = await this.getToken();
+        let hasSubscription = false;
+        if (getToken) {
+            this.requestOptions.headers.set('authorization', 'Bearer ' + getToken);
+            try {
+                let suburl = `/users/me/subscription/status`;
+                let uri = new URL(suburl, this.apiurl);
+                let request = new Request(uri, this.requestOptions);
+                let subdata = await this.fetchJSON(request);
+                hasSubscription = subdata.subscription_service_status == "enabled";
+            } catch (_e) {
+                //page will give 401 if not loggedin
+                hasSubscription = false;
+            }
+        }
         let nextCursor = `/episodes?comic_id=${manga.id}&order=asc&sort=episode_order`;
-        for(let run = true; run;) {
-            const data = await this._getChaptersFromPage(manga, nextCursor);
+        if (hasSubscription) {
+            nextCursor += "&episode_status=free_viewing%2Conly_for_subscription";
+        }
+        for (let run = true; run;) {
+            const data = await this._getChaptersFromPage(manga, nextCursor, hasSubscription);
             nextCursor = data.nextCursor;
             chapterList.push(...data.chapters);
-            if(nextCursor == null) {
+            if (nextCursor == null) {
                 run = false;
             }
         }
         return chapterList.reverse();
     }
+    //get auth token from indexedDB
+    async getToken() {
+        let script = `new Promise(resolve => {
```

Comment:
The cyclomatic complexity is quite high for this script, may try to reduce nested levels